### PR TITLE
Test Cases Added in test_ode.py

### LINKED ISSUE #12623: Weird results for 2nd order ode
Hello,

I was using sympy ode to solve good ol' RLC with excitation term. Depending on the constant of the excitation, I sometimes get very weird results.

`from sympy import *
from sympy.interactive import printing
printing.init_printing(use_latex=True)

t = symbols("t")
u = symbols("u",cls=Function)
R,L,C,E_0 = symbols("R L C E_0",positive=True)`

`eqRLC = Eq( L*C*u(t).diff(t,t) + R*C*u(t).diff(t) + u(t),E_0)
dsolve(eqRLC)`

This is fine. I do not get the critical case where R = 2 sqrt(L/C), but as a physicist I do not really care. Then I divide my equation by LC. dsolve takes a long time and I the results are...

`eqRLC_2 = Eq( u(t).diff(t,t) + R /L*u(t).diff(t) + 1/(L*C)*u(t),E_0/(L*C))
dsolve(eqRLC_2)`
rlc_error

Of course _something_ happens when C*R*R - 4*L = 0, and the final result might even be true, but should not look that awful. The problem comes from the LC in the source term : juste using another constant alpha = E_0/(L*C) make the problem disappear:

`eqRLC3 = Eq( u(t).diff(t,t) + R /L*u(t).diff(t) + 1/(L*C)*u(t), alpha)
dsolve(eqRLC3)`
rlc_error2

Thank for all your work!

### PR DESCRIPTION
**Pull request by Psycho-Pirate**
#### References to other Issues or PRs

Partial fix for #12623 

#### Brief description of what is fixed or changed
Test cases are added in test_ode.py

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/solvers/ode/tests/test_ode.py [hunk 1]
```diff
@@ -2566,6 +2566,22 @@ def test_nth_linear_constant_coeff_undetermined_coefficients():
     assert checkodesol(eq27, sol27, order=2, solve_for_func=False)[0]
     assert checkodesol(eq28, sol28, order=1, solve_for_func=False)[0]
 
+def test_issue_12623():
```
Comment (oscarbenjamin) [inline sympy/solvers/ode/tests/test_ode.py]: There should be two blank lines above and below this function

**Comment by Psycho-Pirate** [comment]
Please review @oscarbenjamin .

**Comment by oscarbenjamin** [comment]
I think that these test cases are good but on reflection this does not completely close the issue so I've edited the OP to say "partial fix".

This looks good. Merging. Thanks for this!

**Comment by Psycho-Pirate** [comment]
Thanks for reviewing! @oscarbenjamin

**Pull request merged**